x-pack/filebeat/input/http_endpoint/httpserver.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpendpoint
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+type HttpServer struct {
+	server *http.Server
+	ctx    context.Context
+	stop   context.CancelFunc
+	done   chan struct{}
+}
+
+func (h *HttpServer) Start() error {
+	go func() {
+		if h.server.TLSConfig != nil {
+			logp.Info("Starting HTTPS server on %s", h.server.Addr)
```

Comment:
Please add a logger to `HttpServer`, so you can avoid using `logp.Info`. We are trying to get rid of these global functions ATM.